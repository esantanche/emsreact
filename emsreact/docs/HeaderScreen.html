<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: HeaderScreen</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: HeaderScreen</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>HeaderScreen<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description"><p>This is the header always present at the top of the page.</p>
<p>There are the home button (the text &quot;Emanuele Santanch√©, Web Developer&quot;) and the menu button.</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="HeaderScreen"><span class="type-signature"></span>new HeaderScreen<span class="signature">()</span><span class="type-signature"></span></h4>
    

    















<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="header_HeaderScreen.js.html">header/HeaderScreen.js</a>, <a href="header_HeaderScreen.js.html#line30">line 30</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="handleClickAwayFromMenu"><span class="type-signature"></span>handleClickAwayFromMenu<span class="type-signature"></span></h4>




<div class="description">
    <p>Called when the user clicks somewhere outside the menu.</p>
<p>It means that we have to close the menu.</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="header_HeaderScreen.js.html">header/HeaderScreen.js</a>, <a href="header_HeaderScreen.js.html#line90">line 90</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleMenuButtonClick"><span class="type-signature"></span>handleMenuButtonClick<span class="type-signature"></span></h4>




<div class="description">
    <p>When the menu button, the button made of three lines, il clicked,
this function is called and we have to show the menu</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="header_HeaderScreen.js.html">header/HeaderScreen.js</a>, <a href="header_HeaderScreen.js.html#line52">line 52</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="handleMenuClose"><span class="type-signature"></span>handleMenuClose<span class="type-signature"></span></h4>




<div class="description">
    <p>Called when an item of the menu is clicked.</p>
<p>There may be a url to go to or we may have to show the Pick My Brain
dialog.</p>
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="header_HeaderScreen.js.html">header/HeaderScreen.js</a>, <a href="header_HeaderScreen.js.html#line70">line 70</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="handleOnClickPickMyBrainDialog"><span class="type-signature"></span>handleOnClickPickMyBrainDialog<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Called when the user clicks on the &quot;X&quot; to close the dialog.</p>
<p>We close the dialog by setting pick_my_brain_dialog_is_open to false.</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="header_HeaderScreen.js.html">header/HeaderScreen.js</a>, <a href="header_HeaderScreen.js.html#line100">line 100</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AlignedGridPane.html">AlignedGridPane</a></li><li><a href="App.html">App</a></li><li><a href="ArticleCard.html">ArticleCard</a></li><li><a href="ArticleScreen.html">ArticleScreen</a></li><li><a href="ArticleService.html">ArticleService</a></li><li><a href="BodyPane.html">BodyPane</a></li><li><a href="BottomGridPane.html">BottomGridPane</a></li><li><a href="ColouredCard.html">ColouredCard</a></li><li><a href="FooterScreen.html">FooterScreen</a></li><li><a href="FrontGridPane.html">FrontGridPane</a></li><li><a href="HeaderMenu.html">HeaderMenu</a></li><li><a href="HeaderScreen.html">HeaderScreen</a></li><li><a href="HomeButtonPane.html">HomeButtonPane</a></li><li><a href="HomeButtonText.html">HomeButtonText</a></li><li><a href="HomeScreen.html">HomeScreen</a></li><li><a href="InnerGridPane.html">InnerGridPane</a></li><li><a href="ListOfArticlesScreen.html">ListOfArticlesScreen</a></li><li><a href="MasterAppBar.html">MasterAppBar</a></li><li><a href="MasterAvatar.html">MasterAvatar</a></li><li><a href="MenuButton.html">MenuButton</a></li><li><a href="MessageDialog.html">MessageDialog</a></li><li><a href="PickMyBrainDialog.html">PickMyBrainDialog</a></li><li><a href="PickMyBrainMessageService.html">PickMyBrainMessageService</a></li><li><a href="PickMyBrainScreen.html">PickMyBrainScreen</a></li><li><a href="SeparatorPane.html">SeparatorPane</a></li><li><a href="SocialCard.html">SocialCard</a></li><li><a href="SocialScreen.html">SocialScreen</a></li><li><a href="TopicScreen.html">TopicScreen</a></li><li><a href="TopicService.html">TopicService</a></li></ul><h3>Global</h3><ul><li><a href="global.html#theme">theme</a></li><li><a href="global.html#title_to_slug">title_to_slug</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Tue Aug 21 2018 07:30:21 GMT+0000 (UTC)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>