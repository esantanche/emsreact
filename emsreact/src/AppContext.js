// This is the application context
// We populate it in App.js and provide it to the entire application

import React from "react";

const AppContext = React.createContext(null);

export default AppContext;