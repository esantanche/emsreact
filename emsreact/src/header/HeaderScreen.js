import React, { Component } from 'react';

// Material UI components
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';

// From components library
import MasterAppBar from "../components_library/bars/MasterAppBar";
import AlignedGridPane from "../components_library/panes/AlignedGridPane";
import HomeButtonPane from "../components_library/panes/HomeButtonPane";
import HomeButtonText from "../components_library/texts/HomeButtonText";
import HeaderMenu from "../components_library/menus/HeaderMenu";
import PickMyBrainDialog from "../components_library/dialogs/PickMyBrainDialog";
import MenuButton from "../components_library/buttons/MenuButton";
import FrontGridPane from "../components_library/panes/FrontGridPane";
import MenuButtonPane from "../components_library/panes/MenuButtonPane";

// Needed to use history and move to other url
import {withRouter} from "react-router-dom";

// Material UI function that make width available
import withWidth from "@material-ui/core/withWidth/withWidth";

/**
 * This is the header always present at the top of the page.
 *
 * There are the home button (the text "Emanuele Santanché, Web Developer") and the menu button.
 */
class HeaderScreen extends Component {

    constructor(props) {
        super(props);

        this.state = {
            // The element the menu is anchored to, typically a button that opens the menu
            // When anchorElement is null, the menu is invisible
            anchorElement: null,
            // The Pick My Brain dialog starts invisible. It will become
            // visible when the user will click on "Pick My Brain!" in the menu
            pick_my_brain_dialog_is_open: false
        };

    }

    /**
     * When the menu button, the button made of three lines, il clicked,
     * this function is called and we have to show the menu
     *
     * @param {object} event Click event for menu button
     */
    handleMenuButtonClick = event => {

        // By setting the anchor element for the menu to the menu button
        // the user has just clicked, we show the menu

        this.setState({ anchorElement: event.currentTarget });
    };

    /**
     * Called when an item of the menu is clicked.
     *
     * There may be a url to go to or we may have to show the Pick My Brain
     * dialog.
     *
     * @param {string} url_to_go_to The url we have to go to
     * @param {boolean} open_pick_my_brain_dialog True if the user clicked on
     * "Pick My Brain!" and we have to show the dialog PickMyBrainDialog
     */
    handleMenuClose = (url_to_go_to, open_pick_my_brain_dialog) => {

        this.setState({ anchorElement: null });

        if (url_to_go_to) {

            this.props.history.push(url_to_go_to);

        } else if (open_pick_my_brain_dialog) {

            this.setState({ pick_my_brain_dialog_is_open: true });

        }
    };

    /**
     * Called when the user clicks somewhere outside the menu.
     *
     * It means that we have to close the menu.
     */
    handleClickAwayFromMenu = () => {

        this.setState({ anchorElement: null });
    };

    /**
     * Called when the user clicks on the "X" to close the dialog.
     *
     * We close the dialog by setting pick_my_brain_dialog_is_open to false.
     */
    handleOnClickPickMyBrainDialog() {

        this.setState({ pick_my_brain_dialog_is_open: false });
    };

    render() {

        const { width } = this.props;

        return (

            <React.Fragment>

                <FrontGridPane>

                    <MasterAppBar>

                        <AlignedGridPane>
                            <Grid item xs={11}>
                                <HomeButtonPane>
                                    <Button href="/">

                                        {
                                            width === 'xs' ?
                                                <HomeButtonText>
                                                    Emanuele Santanché,<br /> Web Developer
                                                </HomeButtonText>
                                                :
                                                <HomeButtonText>
                                                    Emanuele Santanché, Web Developer
                                                </HomeButtonText>
                                        }

                                    </Button>
                                </HomeButtonPane>
                            </Grid>

                            <Grid item xs={1}>
                                <MenuButtonPane>
                                    <MenuButton onclick={this.handleMenuButtonClick.bind(this)} />
                                </MenuButtonPane>
                                <ClickAwayListener onClickAway={this.handleClickAwayFromMenu.bind(this)}>
                                    <HeaderMenu anchorElement={this.state.anchorElement}
                                                handleMenuClose={this.handleMenuClose.bind(this)}/>
                                </ClickAwayListener>
                            </Grid>
                        </AlignedGridPane>

                    </MasterAppBar>

                </FrontGridPane>

                <PickMyBrainDialog open={this.state.pick_my_brain_dialog_is_open}
                                   onclick={this.handleOnClickPickMyBrainDialog.bind(this)}/>

            </React.Fragment>

        )
    }

}

// withRouter is needed to be able to use the history
// like in:
//    this.props.history.push(url_to_go_to);
export default withRouter(withWidth()(HeaderScreen));

