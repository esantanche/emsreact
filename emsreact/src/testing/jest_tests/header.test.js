// FIXME docs? anything to do?
// Testing everything in folder article
// fixme fix docs

import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import HeaderScreen from "../../header/HeaderScreen";
import {BrowserRouter} from "react-router-dom";
import AppContext from "../../AppContext";

//const AppArticleService = new ArticleService();
// const AppTopicService = new TopicService();

// const Services = { TopicService: null,
//     ArticleService: AppArticleService,
//     PickMyBrainMessageService: null };


// FIXME this is for the footer


//                            <FooterScreen />

// const json_response_rest_api_call = [
//     {
//         "nid": "58",
//         "title": "Privacy Policy",
//         "body": "This is the body of the privacy policy. It\' for testing",
//         "langcode": "English",
//         "field_ems_topic": "Small print",
//         "sticky": "Off",
//         "field_image": ""
//     }
// ];

// const article_screen_to_be_tested = ( <AppContext.Provider value={Services}>
//                                          <ArticleScreen match={{ params: { articlenid: 58 } }}
//                                          />
//                                       </AppContext.Provider> );

describe('Testing HeaderScreen', function() {


    // const AppPickMyBrainMessageService = new PickMyBrainMessageService();
    //
    // const Services = { TopicService: null,
    //     ArticleService: null,
    //     PickMyBrainMessageService: AppPickMyBrainMessageService };

    // const Services = { TopicService: AppTopicService,
    //     ArticleService: AppArticleService,
    //     PickMyBrainMessageService: null };

    // const header_screen_to_be_tested = ( <AppContext.Provider value={Services}>
    //                                         <HeaderScreen />
    //                                     </AppContext.Provider> );

    const header_screen_to_be_tested = ( <BrowserRouter>
                                            <HeaderScreen />
                                        </BrowserRouter> );

    // const footer_screen_to_be_tested = (  );

    // const message_dialog_confirmation_message_to_be_tested = ( <MessageDialog open={true} it_is_an_error_message={false}
    //                                                                           title="Message sent"
    //                                                                           message="Thank you for your message!"/> );

    it('HeaderScreen renders correctly', () => {

        // const tree = shallow(header_menu_to_be_tested);
        //
        // expect(EnzymeToJson(tree)).toMatchSnapshot();
        const tree = renderer
            .create(header_screen_to_be_tested)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

});
