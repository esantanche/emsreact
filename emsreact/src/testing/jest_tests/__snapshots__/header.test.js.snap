// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HeaderScreen HeaderScreen renders correctly 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-16-23"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-md-1-57 MuiGrid-grid-lg-2-72"
  />
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-40 MuiGrid-grid-md-10-66 MuiGrid-grid-lg-8-78"
  >
    <header
      className="MuiPaper-root-106 MuiPaper-elevation4-112 MuiAppBar-root-98 MuiAppBar-positionSticky-101 MuiAppBar-colorPrimary-104 MasterAppBar-MasterAppBar-97"
    >
      <div
        className="MuiGrid-container-1 AlignedGridPane-AlignedGridPane-133"
      >
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-11-39"
        >
          <div
            className="HomeButtonPane-HomeButtonPane-180 HomeButtonPane-TopAndBottomMargins-181"
          >
            <a
              className="MuiButtonBase-root-159 MuiButton-root-135 MuiButton-text-137 MuiButton-flat-140"
              href="/"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex="0"
            >
              <span
                className="MuiButton-label-136"
              >
                <h1
                  className="MuiTypography-root-195 MuiTypography-headline-200 HomeButtonText-HomeButtonText-194"
                >
                  Emanuele Santanch√©, Web Developer
                </h1>
              </span>
              <span
                className="MuiTouchRipple-root-169"
              />
            </a>
          </div>
        </div>
        <div
          className="MuiGrid-item-2 MuiGrid-grid-xs-1-29"
        >
          <button
            className="MuiButtonBase-root-159 MuiButton-root-135 MuiButton-text-137 MuiButton-flat-140 MenuButton-button-134"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-136"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-162"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-169"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-md-1-57 MuiGrid-grid-lg-2-72"
  />
</div>
`;
